src/query-ast-to-sql-ast/index.js

```diff
@@ -186,6 +186,16 @@ export function populateASTNode(
     // we'll set a flag for pagination.
     if (fieldConfig.sqlPaginate) {
       sqlASTNode.paginate = true
+      if (fieldConfig.sqlPageLimit) {
+        if (
+          fieldConfig.sqlPageLimit < sqlASTNode.args.first ||
+          fieldConfig.sqlPageLimit < sqlASTNode.args.last
+        ) {
+          throw new Error(
+            `Maximum page size of ${gqlType.name} type, is ${fieldConfig.sqlPageLimit}`
+          )
+        }
+      }
```

Comment:
This accounts for the case where the query specifies a `first` or `last` argument, but if the query doesn't specify an argument at all, then this condition will never be triggered, allowing a query to fetch *all* the data without any page size at all. I think we should prevent that as that's the kind of performance pathology that is specifically really bad! 

I see two ways forward:
 - default the page size to the `sqlPageLimit` if it is set
 - add another parameter like `sqlDefaultPageSize` which would allow a default smaller than the maximum.
 
Thoughts?

Reply:
the second approach seems better to me, as it helps allow users to have different max and default page size limits. Shall I go ahead and make changes accordingly?

Reply:
added sqlDefaultPageSize field